### PR TITLE
Add tests for completions crash

### LINKED ISSUE #52792: Opening a file causes subsequent completion failures
```ts
// @filename: classes.ts
import { Component } from "./utils.js";

export class MyComponent extends Component {
    render/**/
}

// @filename: utils.ts
export class Element {
    // ...
}

export abstract class Component {
    abstract render(): Element;
}
```

```json5
// @filename: tsconfig.json
{
    "compilerOptions": {
        "target": "es2022",
        "module": "nodenext",
        "strict": true,
        "outDir": "../out",
        "moduleDetection": "force"
    }
}
```

1. Request completions with auto-imports and class member completions at /**/
2. Open up `utils.ts` so that it is not just in the program, but is also an open file.
3. Erase a single character at the marker.
4. Request completions.

```
Response received: completionInfo (32). Request took 54 ms. Success: false . Message: Error processing request. Debug Failure.
Error: Debug Failure.
    at Object.addImportFromExportedSymbol (tsserver.js:141295:13)
    at tsserver.js:147019:38
    at Array.forEach (<anonymous>)
    at importSymbols (tsserver.js:147019:11)
    at createSignatureDeclarationFromSignature (tsserver.js:146653:9)
    at outputMethod (tsserver.js:146541:20)
    at Object.addNewNodeForMemberSymbol (tsserver.js:146523:9)
    at getEntryForMemberCompletion (tsserver.js:148878:22)
    at createCompletionEntry (tsserver.js:148775:64)
    at getCompletionEntriesFromSymbols (tsserver.js:149316:19)
    at completionInfoFromData (tsserver.js:148442:23)
    at Object.getCompletionsAtPosition (tsserver.js:148264:24)
    at Object.getCompletionsAtPosition2 [as getCompletionsAtPosition] (tsserver.js:135381:35)
    at IpcIOSession.getCompletions (tsserver.js:178887:54)
    at completionInfo (tsserver.js:177298:43)
    at tsserver.js:179651:69
    at IpcIOSession.executeWithRequestId (tsserver.js:179643:14)
    at IpcIOSession.executeCommand (tsserver.js:179651:29)
    at IpcIOSession.onMessage (tsserver.js:179693:51)
    at process.<anonymous> (tsserver.js:181261:14)
    at process.emit (node:events:526:28)
    at emit (node:internal/child_process:938:14)
```

### PR DESCRIPTION
**Pull request by andrewbranch**
Closes #52792. It turned out to be fully fixed by #53001.

### CODE REVIEW

File: src/testRunner/unittests/tsserver/exportMapCache.ts [hunk 1]
```diff
@@ -129,6 +129,92 @@ describe("unittests:: tsserver:: exportMapCache", () => {
...
+            command: ts.server.protocol.CommandTypes.CompletionInfo,
+            arguments: {
+                file: classesTs.path,
+                line: 4,
+                offset: 22,
+                prefix: "rende",
+                includeExternalModuleExports: true,
+                includeInsertTextCompletions: true,
+            }
+        });
```
Comment (sheetalkamat) [inline src/testRunner/unittests/tsserver/exportMapCache.ts:216]: Can you please baseline tsserver log so its easier to debug and find out whats going on when  esp some of the stuff about project creation/watches change,

Comment (andrewbranch) [reply src/testRunner/unittests/tsserver/exportMapCache.ts:216]: How do I do that?

Comment (sheetalkamat) [reply src/testRunner/unittests/tsserver/exportMapCache.ts:216]: use createLoggerWithInMemoryLogs and baselineTsserverLogs eg: https://github.com/microsoft/TypeScript/blob/main/src/testRunner/unittests/tsserver/openFile.ts#L42

**Comment by DanielRosenwasser** [comment]
What sorts of follow-up work would still need to be done after #53001?

**Review by sheetalkamat — APPROVED** [review]

**Comment by andrewbranch** [comment]
I don’t know of any follow-up work that needs to be done.

**Pull request merged**